### ISSUE TITLE
Errors while running the script

### ISSUE THREAD

**Issue by kafkaphoenix** [skepticoin/skepticoin#18]
![error2](https://user-images.githubusercontent.com/17835242/117585165-3237a500-b111-11eb-81b9-88c5a8b2fb2a.png)
![error](https://user-images.githubusercontent.com/17835242/117585166-32d03b80-b111-11eb-9ebf-4e441f1b5bf0.png)

The memory error probably is not important as I was running the script in a raspberry all the day and probably the system wasn't enough for it

But the other error happens me after a day running the script and I need to stop the script and start again

**Comment by naeu** [comment]
If a miner X finds a block and broadcasts it, only to receive a block with another solution at the same height one millisecond later, what happens? The miner rejects it and continues mining? Then the miner Y that sent the block rejects the block from miner X and continues mining. Wouldn't that branch the chain?

The only thing preventing this is the quick propagation of blocks across the network combined with the low chance of finding a block. The problem gets worse with more miners at low difficulty. I'm not sure if it's a transaction error or a block error, but someone trying to send coins while on a branched chain might cause that "output_reference does not exist" error to appear.

Shouldn't these numbers match?
```
(cd chain; ls | cut -d '-' -f 1 | sort | wc -l)
(cd chain; ls | cut -d '-' -f 1 | sort -u | wc -l)
```

**Comment by kafkaphoenix** [comment]
Hmmm interesting, I updated one chain folder with chains from other miner's folder in my second pc so maybe that's the reason of the conflict

And the numbers don't match
first one: 63730
second one: 63099

**Comment by sashimi-houdini** [comment]
> the other error

yeah so this is the "INVALID TRANSACTION" error, which signifies exactly that: your peer has been sent an invalid transaction, and refuses to put it in a block. This could be caused by any number of things, such as

* a programming error in `skepticoin`, causing `skepticoin-send` to accidentally create invalid transactions
* people ~~playing around~~ mounting a serious attack on our network

Given that somebody already admitted to the latter in another issue, I'm guessing that's what's happened here. We'll only get confirmation of the former if somebody starts complaining about coin they tried to send not arriving at the destination.

For now, I'm leaving the big fat warning in, so that we get some feedback on what's going on.

**Comment by sashimi-houdini** [comment]
> If a miner X finds a block and broadcasts it, only to receive a block with another solution at the same height one millisecond later, what happens? The miner rejects it and continues mining? Then the miner Y that sent the block rejects the block from miner X and continues mining. Wouldn't that branch the chain?

yes, but only temporarily. At some point, either branch will be longer, at which point miners will continue at the longest branch again. Straight from the Bitcoin whitepaper.

**Comment by sashimi-houdini** [comment]
I'm closing this issue, because it is about a number of unrelated things that have good explanations for them. If one of those warrants further research, let's pick it up in a new issue thread with a nice single subject.
